src/bin/krillup.rs

```diff
@@ -1,89 +1,161 @@
-                if let Err(e) = migrate(config, target) {
-                    eprintln!("*** Error Migrating DATA ***");
+            };
+
+            match prepare_upgrade_data_migrations(
+                UpgradeMode::PrepareOnly,
+                &config,
+                &properties_manager,
+            ) {
+                Err(e) => {
+                    eprintln!("*** Error Preparing Data Migration ***");
                     eprintln!("{}", e);
                     eprintln!();
                     eprintln!(
-                        "Note that your server data has NOT been modified."
+                        "Note that your server data has NOT been modified. \
+                         Do not upgrade krill"
+                    );
+                    eprintln!("itself yet!");
+                    eprintln!(
+                        "If you did upgrade krill, then downgrade it to \
+                         the previous installed version."
                     );
                     ::std::process::exit(1);
                 }
+                Ok(None) => info!("No update needed"),
+                Ok(Some(report)) => {
+                    let from = report.versions().from();
+                    let to = report.versions().to();
+                    if report.data_migration() {
+                        info!(
+                            "Prepared and verified upgrade from {} to {}.",
+                            from, to
+                        );
+                    } else {
+                        info!(
+                            "No preparation is needed for the upgrade from \
+                             {} to {}.",
+                            from, to
+                        )
+                    }
+                },
+            }
+        }
+        Command::Migrate(cmd) => {
+            if let Err(e) = migrate(config, cmd.target) {
+                eprintln!("*** Error Migrating DATA ***");
+                eprintln!("{}", e);
+                eprintln!();
+                eprintln!(
+                    "Note that your server data has NOT been modified."
+                );
+                ::std::process::exit(1);
             }
-        },
+        }
     }
 }
 
+
+//------------ Options -------------------------------------------------------
+
+/// The command line options for the krillup command.
+#[derive(clap::Parser)]
+pub struct Options {
+    /// Path to the Krill config file
+    #[arg(
+        short, long,
+        value_name = "path",
+        default_value = constants::KRILL_DEFAULT_CONFIG_FILE,
+    )]
+    pub config: PathBuf,
+
+    #[command(subcommand)]
+    pub command: Command,
+}
+
+
+//------------ Command -------------------------------------------------------
+
+#[derive(clap::Subcommand)]
+pub enum Command {
+    /// Prepare a migration, leave current data unmodified
+    Prepare(Prepare),
+
+    /// Migrate data to different storage. Stop Krill before use!
```

Comment:
What do you mean here by "different storage"? Migration is about upgrading the stored content to the new serialized format, right?

Reply:
I think this is also intended to later move data into a database? In any case, this is basically just taken from the old code, so I want to leave it.